source/tools/testConsistency/dotnetWPF-renderer/WPFRenderer/WPFRenderer/Program.cs

```diff
@@ -0,0 +1,112 @@
+            resultJson.Add("errors", "");
+            Console.OutputEncoding = Encoding.UTF8;
+            try
+            {
+                StringBuilder inputSb = new StringBuilder();
+                string currLine;
+
+                //read a multi line input from stdin
+                do
+                {
+                    currLine = Console.ReadLine();
+                    inputSb = inputSb.Append(currLine);
+                }
+                while ((currLine != null) && (currLine != ""));
+
+                JObject jsonInput = JObject.Parse(inputSb.ToString());
+                string cardData = jsonInput.GetValue("data").ToString();
+                string cardTemplate = jsonInput.GetValue("template").ToString();
+
+                AdaptiveCardTemplate adaptiveCardTemplate = new AdaptiveCardTemplate(cardTemplate);
+                string adaptiveCardJson = adaptiveCardTemplate.Expand(cardData);
+                AdaptiveCardParseResult parseResult = AdaptiveCard.FromJson(adaptiveCardJson);
+                AdaptiveCardRenderer renderer = new AdaptiveCardRenderer();
+
+                GetResultJsonString(parseResult, renderer, resultJson, warningList).GetAwaiter().GetResult();
+            }
+
+            catch (Newtonsoft.Json.JsonReaderException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("The input is not a valid json");
+            }
+            catch (AdaptiveTemplateException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("Unable to bind the data with the template. The supplied card data is not valid");
+            }
+            catch (AdaptiveSerializationException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("The supplied json input is not a valid Adaptive Card");
+
+            }
+            catch (ArgumentNullException e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("Unable to capture the image of the card. Ensure that the card template and card data are valid and try again.");
+            }
+            catch (Exception e)
+            {
+                errorList.Add("Exception: " + e.Message);
+                errorList.Add("An error was encountered: " + e.Message);
+            }
+
+            finally
+            {
+                resultJson["warnings"] = new JArray(warningList);
+                resultJson["errors"] = new JArray(errorList);
+                Console.WriteLine(resultJson.ToString());
+            }
+        }
+
+        private static async Task GetResultJsonString(AdaptiveCardParseResult parseResult, AdaptiveCardRenderer renderer, JObject resultJson, List<string> warningList)
+        {
+            AdaptiveCard adaptiveCard = parseResult.Card;
+            CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
```

Comment:
>ds(30)) [](start = 88, length = 7)

Nit: hardcoded magic numbers! This should be a configurable value encapsulated in a variable - you can then actually configure this variable to be say 1s to simulate failures in your testing.